src/common/snippets/src/lowered/pass/init_live_ranges.cpp

```diff
@@ -34,19 +34,21 @@ bool InitLiveRanges::run(LinearIR& linear_ir) {
     for (auto expr_it = linear_ir.begin(); expr_it != linear_ir.end(); expr_it++) {
         const auto& expr = *expr_it;
         const auto op = expr->get_node();
-        if (pass_through_expr(expr)) {
-            if (expr_it != linear_ir.begin())
-                expr->set_live_regs(std::prev(expr_it)->get()->get_live_regs());
-            continue;
-        }
         const double start = expr->get_exec_num();
-        // Remove all regs that expired before start
-        regs_to_expire.erase(regs_to_expire.begin(), regs_to_expire.lower_bound(start)); // remove all elements lower than start (not equal)
         std::set<Reg> live_regs;
-        for (const auto& time_reg : regs_to_expire)
-            live_regs.insert(time_reg.second.begin(), time_reg.second.end());
 
```

Comment:
As we discussed offline, may I ask you to leave the comment here with explanation: why we have different ways for `pass_through_expr` and common expressions? Additionally why we process `pass_through_expr` again if we already processed them before

Reply:
Added the comment

Reply:
And also used `std::move` for live regs in `then` branch